benches/raw_insertions.rs

```diff
@@ -0,0 +1,32 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use rand::Rng;
+use std::hint::black_box;
+use vex_prover::imt::order::Order;
+use vex_prover::imt::IndexedMerkleTree;
+use vex_prover::types::{Price, Time, Volume};
+
+fn insert_batch(n: u32) {
+    let mut imt = IndexedMerkleTree::new();
+    let mut rng = rand::thread_rng();
+    let orders = (0..n)
+        .map(|i| {
+            Order::new(
+                Volume::from_u64(rng.gen()),
+                Price::from_u64(rng.gen()),
+                Time::from_u64((i + 2) as u64),
+            )
+        })
+        .collect::<Vec<_>>();
+    for order in orders {
+        imt.insert(order);
+    }
```

Comment:
The current implementation has a few potential issues:

1. Sequential time assignment may not reflect real-world scenarios where orders can arrive with varying timestamps.
2. Collecting all orders into a vector before insertion could impact memory usage for large batches.

Maybe something like this:
```suggestion
    for _ in 0..n {
         let order = Order::new(
             Volume::from_u64(rng.gen()),
             Price::from_u64(rng.gen()),
             Time::from_u64(rng.gen()),
         );
         imt.insert(order);
     }
```

Or you can generate the time in a tighter range, ensuring no overlap

Reply:
The circuit assumes that the orders are entered in sequence. i.e oi.time < oi+1.time , we must generate random numbers that that is bigger than those generated before for time

Reply:
why does the circuit assume orders are entered in sequence? The indexed merkle tree woulnt care, would it?

Reply:
it would, it instantly matches an order that satisfies the price time priority, consider a scenario where we dont assume the orders are in sequnce, consider the orders
1) price p, time t is in the sequnce at ith index,
2) price p, time t - 1 is in the sequence at i + 1 th index
the first order is ahead in the queue which will be matched instantly, where ideally the 2nd order must be matched